### PR TITLE
added initial comment event handler

### LINKED ISSUE #22208: Add a `/check-required-labels` command to force the require-matching-label plugin to reprocess a PR.
**What would you like to be added**:

Add a `/check-require-labels` command to force the `require-matching-label` plugin to reprocess a PR. Please also remember to include the plugin help information for the new command. (e.g.)[https://github.com/kubernetes/test-infra/blob/3f22d33e92bc42bc5c24eacbe4648ab8cb0ac63d/prow/plugins/cla/cla.go#L75-L81]

**Why is this needed**:

Currently the [`require-matching-label` plugin](https://github.com/kubernetes/test-infra/tree/master/prow/plugins/require-matching-label) only processes a PR to check that the `missing_label` is properly applied or removed when it sees webhook events that could be relevant. This means that missing/dropping a webhook can cause a PR to end up in an inconsistent state.

Context: https://github.com/kubernetes/test-infra/issues/22139
/help-wanted
/area prow/plugins

This command will be pretty similar to the [`/check-cla`](https://github.com/kubernetes/test-infra/blob/3f22d33e92bc42bc5c24eacbe4648ab8cb0ac63d/prow/plugins/cla/cla.go#L61) command so I this should be a good first issue.
/good-first-issue

### PR DESCRIPTION
**Pull request by rahulii**
Fixes #22208 
This PR adds the support for `/check-required-labels` command

### CODE REVIEW

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 1]
```diff
@@ -54,6 +54,11 @@ const (
 	pluginName = "require-matching-label"
 )
 
+var (
+	checkRequiredLablesRe = regexp.MustCompile(`(?mi)^/check-required-labels\s*$`)
```
Comment (petr-muller) [inline prow/plugins/require-matching-label/require-matching-label.go:58]:
```suggestion
	checkRequiredLabelsRe = regexp.MustCompile(`(?mi)^/check-required-labels\s*$`)
```

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 2]
```diff
@@ -95,14 +101,24 @@ func helpProvider(config *plugins.Configuration, _ []config.OrgRepo) (*pluginhel
...
 				"": fmt.Sprintf("The plugin has the following configurations:\n<ul><li>%s</li></ul>", strings.Join(descs, "</li><li>")),
 			},
 			Snippet: yamlSnippet,
-		},
-		nil
+		}
+	
+	pluginHelp.AddCommand(pluginhelp.Command{
+			Usage:       "/check-required-labels",
+			Description: "Forces the require-matching-label plugin to reprocess an issue or PR that do not have any labels matching.An example of this is applying a 'needs-sig' label to all issues that do not have a 'sig/*' label",
```
Comment (petr-muller) [inline prow/plugins/require-matching-label/require-matching-label.go:114]:
```suggestion
			Description: "Forces the require-matching-label plugin to reprocess an issue or PR that do not have any labels matching. An example of this is applying a 'needs-sig' label to all issues that do not have a 'sig/*' label",
```

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 3]
```diff
@@ -155,6 +171,22 @@ func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
...
+	//consider open and close PRs and new comments
+	if e.IssueState != "open" || e.IssueState != "closed" || e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	// Only consider "/check-required-labels" comments.
+	if !checkRequiredLablesRe.MatchString(e.Body) {
+		return nil
+	}
+
+	//to be continued
```
Comment (petr-muller) [inline prow/plugins/require-matching-label/require-matching-label.go:187]: I assume you will continue in this PR? Otherwise I don't think we should add help/example for a command that does not do anything (yet).

**Review by petr-muller — COMMENTED** [review]
/ok-to-test

**Comment by cjwagner** [comment]
Please make this PR compile, pass existing tests, and add test cases for the new functionality (see this comment for a similar example: https://github.com/kubernetes/test-infra/pull/22258#pullrequestreview-664999343)

**Review by rahulii — COMMENTED** [review]
> functionality

Ah , @cjwagner , I

> Please make this PR compile, pass existing tests, and add test cases for the new functionality (see this comment for a similar example: [#22258 (review)](https://github.com/kubernetes/test-infra/pull/22258#pullrequestreview-664999343))

Ah , @cjwagner I'm not sure about the changes implemented , let me know what you think of it , and then i can write tests as well!

> Please make this PR compile, pass existing tests, and add test cases for the new functionality (see this comment for a similar example: [#22258 (review)](https://github.com/kubernetes/test-infra/pull/22258#pullrequestreview-664999343))

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 4]
```diff
@@ -155,6 +171,22 @@ func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
...
+func handleComment(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) error {
+	//consider open and close PRs and new comments
+	if e.IssueState != "open" || e.IssueState != "closed" || e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	// Only consider "/check-required-labels" comments.
+	if !checkRequiredLablesRe.MatchString(e.Body) {
+		return nil
+	}
+
```
[the remark this answers is not in the document]
Comment (petr-muller) [reply prow/plugins/require-matching-label/require-matching-label.go:184]: I don't think `GenericCommentEvent` is missing any required fields? Comments are not label events so you'll be missing that but the comments indicate the code is prepared to handle their absence just fine.

[the remark this answers is not in the document]
Comment (rahulii) [reply prow/plugins/require-matching-label/require-matching-label.go:184]: Makes sense!
So should i write tests and check if it works properly?

[the remark this answers is not in the document]
Comment (petr-muller) [reply prow/plugins/require-matching-label/require-matching-label.go:184]: Yes, the code definitely needs tests and scenarios with empty labels should be covered.

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 5]
```diff
@@ -54,6 +53,10 @@ const (
 	pluginName = "require-matching-label"
 )
 
+var (
+	checkRequiredLabelsRe = regexp.MustCompile(`(?mi)^/check-required-labels\s*$`)
+)
+
```
Comment (taragu) [inline prow/plugins/require-matching-label/require-matching-label.go:59]: nit: can we add this into the `var` section on line 36?

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 6]
```diff
@@ -95,14 +99,24 @@ func helpProvider(config *plugins.Configuration, _ []config.OrgRepo) (*pluginhel
...
+		Config: map[string]string{
+			"": fmt.Sprintf("The plugin has the following configurations:\n<ul><li>%s</li></ul>", strings.Join(descs, "</li><li>")),
 		},
-		nil
+		Snippet: yamlSnippet,
+	}
+
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/check-required-labels",
+		Description: "Forces the require-matching-label plugin to reprocess an issue or PR that do not have any labels matching.An example of this is applying a 'needs-sig' label to all issues that do not have a 'sig/*' label",
```
Comment (taragu) [inline prow/plugins/require-matching-label/require-matching-label.go:112]: Others please feel free to chime in. I'm still not sure about this description:
```suggestion
              Description: "Forces the require-matching-label plugin to re-check an issue or PR to add any labels it's missing. An example of this is applying a 'needs-sig' label to all issues that do not have a 'sig/*' label",
```

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 7]
```diff
@@ -155,6 +169,32 @@ func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
 	}
 	return handle(pc.Logger, pc.GitHubClient, cp, pc.PluginConfig.RequireMatchingLabel, e)
 }
+func handleCommentEvent(pc plugins.Agent, ce github.GenericCommentEvent) error {
+	return handleComment(pc, &ce)
+}
+func handleComment(pc plugins.Agent, ce *github.GenericCommentEvent) error {
+	//consider open and close PRs and new comments
```
Comment (taragu) [inline prow/plugins/require-matching-label/require-matching-label.go:176]:
```suggestion
      // Consider open and close PRs and new comments
```

File: prow/plugins/require-matching-label/require-matching-label.go [hunk 8]
```diff
@@ -155,6 +169,32 @@ func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
...
+	if ce.IssueState != "open" || ce.IssueState != "closed" || ce.Action != github.GenericCommentActionCreated {
```
Comment (taragu) [inline prow/plugins/require-matching-label/require-matching-label.go:177]: Should it be `(ce.IssueState != "open" && ce.IssueState != "closed") || ce.Action != github.GenericCommentActionCreated`?

Comment (taragu) [reply prow/plugins/require-matching-label/require-matching-label.go:177]: Can we also add some unit tests under `prow/plugins/require-matching-label/require-matching-label_test.go`?
